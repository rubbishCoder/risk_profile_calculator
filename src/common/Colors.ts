import * as React from 'react';

const Colors = {
    red: '#fe0133',
    white: '#ffffff',
    grey: '#808080',
    blue: '#1E90FF',
    black: '#000000',
    black2: '#171717',
    black3: '#3b393a',
    green: '#5cb85c',
    yellow: '#E9D502'
}

export default Colors;