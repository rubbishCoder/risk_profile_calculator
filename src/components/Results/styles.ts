import { StyleSheet } from 'react-native';
import Colors from '../../common/Colors';
import RV from '../../common/utils/RV';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    text: {
        color: Colors.white,
        fontSize: RV(24),
        textAlign: 'center',
        padding: RV(16),
        fontWeight: 'bold'
    },
    whiteCircle: {
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: Colors.white,
        width: RV(200),
        height: RV(200),
        borderRadius: RV(500)
    },
    buttonContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: Colors.black3,
        paddingVertical: RV(16),
        paddingHorizontal: RV(20),
        borderRadius: RV(12)
    },
    buttonText: {
        fontSize: RV(24),
        textAlign: 'center',
        color: Colors.white,
        fontWeight: 'bold'
    },
});

export default styles
