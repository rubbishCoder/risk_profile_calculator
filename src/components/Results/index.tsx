import { StackActions, useNavigation } from '@react-navigation/native';
import React, { useMemo } from 'react';
import { SafeAreaView, Text, TouchableOpacity, View } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import Colors from '../../common/Colors';
import RV from '../../common/utils/RV';
import styles from './styles';
import { resetQuestionDataState } from '../../reducers/questionsData';

const Results = () => {
  const navigation = useNavigation<any>();
  const dispatch = useDispatch<any>();

  const riskProfileScore = useSelector((state: any) => state.questionsData.riskProfileScore)

  const resultData = useMemo(() => {
    if (riskProfileScore >= 80 && riskProfileScore <= 100) {
      return {
        text: `Congrats, you have got great\n'Risk profile score'.`,
        color: Colors.green
      }
    }
    if (riskProfileScore > 50 && riskProfileScore < 80) {
      return {
        text: `Your 'Risk profile score' is not that great, need to improve.`,
        color: Colors.yellow
      }
    }
    if (riskProfileScore <= 50) {
      return {
        text: `Your 'Risk profile score' is very poor, need to work more on improving the score.`,
        color: Colors.red
      }
    }
  }, [riskProfileScore])

  return (
    <SafeAreaView style={{ ...styles.container, backgroundColor: resultData?.color }}>
      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <View style={styles.whiteCircle}>
          <Text style={{ fontSize: RV(60), fontWeight: 'bold', color: resultData?.color }}>{riskProfileScore}%</Text>
        </View>
        <Text style={styles.text}>
          {resultData?.text}
        </Text>
        <TouchableOpacity style={styles.buttonContainer} activeOpacity={1}
          onPress={() => {
            dispatch(resetQuestionDataState())
            navigation.dispatch(StackActions.replace('Assessment'));
          }}>
          <Text style={styles.buttonText}>Re-Evaluate</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  )
}

export default Results;
