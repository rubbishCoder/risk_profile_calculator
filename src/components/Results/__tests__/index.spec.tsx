import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import * as Redux from 'react-redux';
import Results from '..';
import Colors from '../../../common/Colors';

let mockedNavigationDispatch = jest.fn();
let mockedStoreDispatch: any = jest.fn(() => null);

jest.mock('@react-navigation/native', () => {
    return {
        useNavigation: () => ({
            dispatch: mockedNavigationDispatch
        }),
        StackActions: {
            replace: jest.fn().mockImplementation(x => ({ ...x, "type": "Navigation/REPLACE" }))
        }
    };
});

jest.mock('react-redux', () => {
    return {
        useDispatch: jest.fn(),
        useSelector: jest.fn()
    };
});

describe('Results', () => {
    beforeEach(() => {
        mockedNavigationDispatch.mockClear();
    });

    it('verify snapshot', () => {
        const ResultsScreen = render(<Results />);
        const ResultsScreenJson = ResultsScreen.toJSON();
        expect(ResultsScreenJson).toMatchSnapshot();
    });


    it(`renders 'very good message' if 100%`, () => {
        jest.spyOn(Redux, 'useSelector').mockReturnValue(100);
        const ResultsScreen: any = render(<Results />);
        const messageComponent = ResultsScreen.getByText(`Congrats, your 'Risk profile score' is very good.`)
        expect(messageComponent).toBeTruthy();
        expect(ResultsScreen.root.props.style.backgroundColor).toEqual(Colors.green)
    })

    it(`renders 'good message' if 80% to 99%`, () => {
        jest.spyOn(Redux, 'useSelector').mockReturnValue(85);
        const ResultsScreen = render(<Results />);
        const messageComponent = ResultsScreen.getByText(`Congrats, your 'Risk profile score' is good.`)
        expect(messageComponent).toBeTruthy();
        expect(ResultsScreen.root.props.style.backgroundColor).toEqual(Colors.green)
    })

    it(`renders 'not good message' if 51% to 79%`, () => {
        jest.spyOn(Redux, 'useSelector').mockReturnValue(55);
        const ResultsScreen = render(<Results />);
        const messageComponent = ResultsScreen.getByText(`Your 'Risk profile score' is not that good, need to improve.`)
        expect(messageComponent).toBeTruthy();
        expect(ResultsScreen.root.props.style.backgroundColor).toEqual(Colors.yellow)
    })

    it(`renders 'very poor' message if 100%`, () => {
        jest.spyOn(Redux, 'useSelector').mockReturnValue(35);
        const ResultsScreen = render(<Results />);
        const messageComponent = ResultsScreen.getByText(`Your 'Risk profile score' is very poor, need to work more on improving the score.`)
        expect(messageComponent).toBeTruthy();
        expect(ResultsScreen.root.props.style.backgroundColor).toEqual(Colors.red)
    })

    it(`navigate on re-evaluate`, () => {
        jest.spyOn(Redux, 'useDispatch').mockImplementation(() => mockedStoreDispatch);
        const ResultsScreen = render(<Results />);
        const reEvaluateComponent = ResultsScreen.getByText(`Re-Evaluate`)
        expect(reEvaluateComponent).toBeTruthy();
        fireEvent.press(reEvaluateComponent);
        expect(mockedNavigationDispatch).toHaveBeenCalled();
    })
})