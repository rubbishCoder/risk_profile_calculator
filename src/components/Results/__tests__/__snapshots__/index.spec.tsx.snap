// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results verify snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 720,
          "height": 288,
          "justifyContent": "center",
          "width": 288,
        }
      }
    >
      <Text
        style={
          {
            "color": undefined,
            "fontSize": 86,
            "fontWeight": "bold",
          }
        }
      >
        %
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 35,
          "fontWeight": "bold",
          "padding": 23,
          "textAlign": "center",
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#3b393a",
          "borderRadius": 17,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 29,
          "paddingVertical": 23,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 35,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Re-Evaluate
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
